src/main/java/org/triplea/client/launch/LaunchScreen.java

```diff
@@ -0,0 +1,505 @@
+                .add(JButtonBuilder.builder()
+                    .title("Open Save Game")
+                    .actionListener(() -> {
+                      GameRunner.showSaveGameFileChooser().ifPresent(file -> {
+                        ClientSetting.SELECTED_GAME_LOCATION.save(file.getAbsolutePath());
+
+                        try {
+                          GameData newData = GameDataManager.loadGame(file);
+                          showSinglePlayer(newData);
+                        } catch (IOException e) {
+                          ClientLogger.logError("Failed to load: " + file.getAbsolutePath(), e);
+                        }
+                      });
+
+                    })
+                    .build())
+                .add(Box.createVerticalGlue())
+                .borderEtched()
+                .build())
+            .addCenterIf(gameData != null, () -> mapSetup(gameData))
+            .addSouth(buttonPanel(INSTANCE::showInitialScreen, () -> {
+            }))
+            .build());
+  }
+
+  private static JPanel mapSetup(final GameData gameData) {
+    final GameSelectorModel model = new GameSelectorModel();
+    model.load(gameData, "");
+    // backup current game properties before showing dialog
+
+
+    return JPanelBuilder.builder()
+        .addNorth(
+            JPanelBuilder.builder()
+                .flowLayout()
+                .borderEtched()
+                .add(
+                    JPanelBuilder.builder()
+                        .verticalBoxLayout()
+                        .add(new JLabel("<html><h1>" + gameData.getGameName() + " </h1></html>"))
+                        .add(new JLabel("Version: " + gameData.getGameVersion()))
+                        .add(new JLabel("Round: " + gameData.getSequence().getRound()))
+                        .build())
+                .build())
+        .addCenter(
+            JSplitPaneBuilder.builder()
+                .dividerLocation(240)
+                .addLeft(SwingComponents.newJScrollPane(mapOptionsPanel(gameData)))
+                .addRight(
+                    JTabbedPaneBuilder.builder()
+                        .addTab("Player Selection",
+                            SwingComponents.newJScrollPane(
+                                JPanelBuilder.builder()
+                                    .verticalBoxLayout()
+                                    .addEach(playerSelectionPanel(gameData))
+                                    .build()))
+                        .addTab("Handicap",
+                            SwingComponents.newJScrollPane(
+                                JPanelBuilder.builder()
+                                    .verticalBoxLayout()
+                                    .addEach(bidSelectionPanel(gameData))
+                                    .build()))
+                        .build())
+                .build())
+        .build();
+  }
+
+
+  private static JPanel mapOptionsPanel(final GameData gameData) {
```

Comment:
The code is rough still. For example, we do a lot of computation to get all players in an alliance, this UI code should be more direct, less branchy. The start of this class is a decent example, later on it gets to more recent work and the more complex game staging screen.